cli/src/runner.rs

```diff
@@ -264,3 +306,99 @@ pub async fn run_quarantine(
         quarantine_status: quarantine_results,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bundle::{BundledFile, FileSetType};
+    use test_utils::inputs::get_test_file_path;
+
+    /// Contains 1 failure at 1:00
+    const JUNIT0_FAIL: &str = "test_fixtures/junit0_fail.xml";
+    // Contains 1 pass at 2:00
+    const JUNIT0_PASS: &str = "test_fixtures/junit0_pass.xml";
+    // Contains 1 failure at 3:00 and 1 failure at 5:00
+    const JUNIT1_FAIL: &str = "test_fixtures/junit1_fail.xml";
+    // Contains 2 passes at 4:00
+    const JUNIT1_PASS: &str = "test_fixtures/junit1_pass.xml";
```

Comment:
provides a variety of cases with which to test timestamp interleaving. Mainly the only pairing we care about though is `[JUNIT0_FAIL, JUNIT0_PASS]`, and the rest are just to prevent regressions